candig_federation/operations.py

```diff
@@ -113,6 +117,90 @@ def delete_server(server_id):
     return result, 200
 
 
+def list_external_services():
+    """
+    :return: Dictionary of registered external services.
+    """
+    services = get_registered_external_services()
+    if services is not None:
+        result = map(lambda x: x["service"], services.values())
+        return list(result), 200
+    logger.debug(f"Couldn't list services", connexion.request)
+    return {"message": "Couldn't list services"}, 500
+
+
+async def add_external_service():
+    """
+    :return: Service added.
+    """
+    if not is_site_admin(connexion.request):
+        return {"message": "User is not authorized to POST"}, 403
+    try:
+        req = await connexion.request.json()
+        if req is not None and 'service' in req:
+            new_service = register_external_service(req)
+            if new_service is None:
+                return {"message": f"Service matching {new_service['service']} already present"}, 200
```

Comment:
This doesn't seem possible to trip at the moment; if the new service already exists the code looks like it just overwrites the existing value anyway and returns a non-None value.